diesel_derives/src/util.rs

```diff
@@ -93,7 +93,8 @@ where
 }
 
 pub fn wrap_in_dummy_mod(item: TokenStream) -> TokenStream {
-    // #[allow(unused_qualifications)] can be removed if https://github.com/rust-lang/rust/issues/130277 gets done
+    // allow(unused_qualifications) is here as it hard to unsure the span is correctly set. Should stay until it is
```

Comment:
I would rather prefer auditing the derive crate to ensure all usages of spans are encoded correctly. Yes that's a bit of work but it should fix all these issues at once. 
The alternative is to remove this allow and start searching again if something pops up in some code.

Reply:
That seems like a lot of work for something that's seldom going to be noticed considering the warning is not enabled by default. It looks like it's going to be hard to maintain if we don't have it in our CI.

To make that practical to fix and maintain, it looks like [`syn`'s `Spanned` trait](https://docs.rs/syn/latest/syn/spanned/trait.Spanned.html) should have a `span_mixed` method that constructs this.